fix: route options are not merged (outer filter is merged with inner filter) (#8033)
codeigniter4/CodeIgniter4: user_guide_src/source/incoming/routing.rst

This would handle the URL at **admin/users/list**.

**Filter** option passed to the outer ``group()`` are merged with the inner
``group()`` filter option.
The above code runs ``myfilter:config`` for the route ``admin``, and ``myfilter:config``
and ``myfilter:region`` for the route ``admin/users/list``.

> But other options that you specify in the inner ``group()`` options, the value
> is overwritten.

Reviewer: ```suggestion
Any other overlapping options passed to the inner `group()` will overwrite their values.
```

Reviewer: Eh this is tricky to word, I'm not sure mine is strictly better. Take it or leave it.
